### ISSUE TITLE
Regression: USB-C alt mode external display fails to connect

### ISSUE THREAD

**Issue by Doc-4** [GrapheneOS/os-issue-tracker#3759]
I don't know if it was the last update hotfix or the update right before it that did this, but right now if I plug in a USB-C display, I get an error "can't mirror to display" "use a different cable and try again". I have tested a second cable capable of connecting a DP alt mode display just to be safe, neither work anymore and they did work before just a few days ago, and do currently work when connected to devices other than my Pixel. I also made sure my device was unlocked and USB security policy would have permitted the screen to mirror.

Pixel 8 Pro, Build 2024071200

**Comment by anovak89** [comment]
It's not a single user issue. 
https://discuss.grapheneos.org/d/14122-display-mirroring-stopped-working/5

**Comment by treequell** [comment]
Reproduced by @bobster89 and @randomgosuser on Forum: https://discuss.grapheneos.org/d/14122-display-mirroring-stopped-working

**Comment by randomgosuser** [comment]
> @Doc-4 Try disabling all the relevant developer options, disable developer options itself and reboot.

I am the author of the post on the forum, and I already tries this as another user suggested, but the issues remains. No matter what I do, I can no longer get the "mirror display" pop up to appear when I plug the connector

**Comment by Doc-4** [comment]
At the time I encountered this yesterday, Dev options were disabled and hadn't been enabled recently. I did also reboot and test again at the time, still with them disabled. 

To be thorough, the only options showing enabled if I turn on dev options are Verify Bytecode of debuggable apps, WiFi scan throttling, Mobile data always active, Allow window-level blurs, and Stylus handwriting. To the best of my knowledge, those are enabled by default.

**Comment by randomgosuser** [comment]
> At the time I encountered this yesterday, Dev options were disabled and hadn't been enabled recently. I did also reboot and test again at the time, still with them disabled.
> 
> To be thorough, the only options showing enabled if I turn on dev options are Verify Bytecode of debuggable apps, WiFi scan throttling, Mobile data always active, Allow window-level blurs, and Stylus handwriting. To the best of my knowledge, those are enabled by default.

To add info to the developer options point, I tried all possible combinations of the 3 relevant dev options that may affect this, which to my knowledge are:

- Enable freeform windows
- Force desktop mode
- Enable non-resizable in multi window

All these options are now OFF, and I did not change any other dev option. Rebooted the phone and still nothing, same "use a different cable error".

If this is relevant, I can tell you that Force desktop mode was ON at the time of the update, while I only tried the other two after I started getting the error to see if it fixed anything.

@thestinger Do you want us to reboot the phone, plug the cable and see you a log with the error? Thank you

**Comment by randomgosuser** [comment]
> Yes, logs after a reboot would be helpful. It would likely be even better to use safe mode so the logs are cleaner without a bunch of apps running.

Here is the full report captured immediately after a reboot in safe mode. Let me know if you need more data. Thank you!
[bugreport-akita-AP2A.240705.005-2024-07-14-22-19-58.zip](https://github.com/user-attachments/files/16228715/bugreport-akita-AP2A.240705.005-2024-07-14-22-19-58.zip)

**Comment by Doc-4** [comment]
> Yes, logs after a reboot would be helpful. It would likely be even better to use safe mode so the logs are cleaner without a bunch of apps running.

To be clear, did you want a copy of the system log at the time, or did you want a bug report? I saved the system log after plugging and unplugging on a fresh boot to safe mode but if preferred I can go back and get a bug report if one isn't enough.

**Comment by randomgosuser** [comment]
Update: this morning I went back to the office and, since we use thunderbolt 3 docking stations for company laptops I tried plugging that into the phone. Sometimes the device only started charging, and sometimes the "mirror display" option did pop up.
Never got the "use a different cable error message".
However, even when the "mirror display" option appears, if I select to use it it crashes after a few seconds of use, as the monitor loses all signal and goes black. The only way to restore connection is to disconnect the phone, try plugging it back multiple times until the "mirror display" things pops up again, but still the connection only lasts for a few seconds. 
I also tried the usual developer options combination to see if one of them worked, but the end result is always the same: 3-4 seconds of video output and then all input signal is lost.
Even after display output is lost, the phone screen still shows the external display icon in the system tray.

I had never tried this docking station with my phone before, so maybe is just incompatible hardware. Still, since this is the first time that I am able to get some sort of output after the update, I thought to post this.
I also captured a second full bug report in safe mode recording this (immediately after a reboot, developer options OFF): 
[bugreport-akita-AP2A.240705.005-2024-07-15-08-04-12.zip](https://github.com/user-attachments/files/16230776/bugreport-akita-AP2A.240705.005-2024-07-15-08-04-12.zip)

**Comment by treequell** [comment]
@maade93791 Thank you for the fix. Is the fix specific to screen mirroring, or could it also be relevant to audio output via USB-C? I have a user on Forum reporting no audio output via USB-C since updating to 2024071200.

See: https://discuss.grapheneos.org/d/14122-display-mirroring-stopped-working/23

**Comment by treequell** [comment]
> @treequell I'm not sure. Didn't test that.

Appears the issue also affected audio output not only screen mirroring since the user is reporting their issue with audio output says it has been fixed on 2024071600.

**Comment by J0ergL** [comment]
Hello and Thank you so much for this wonderful ROM!
I am switching from  droidcr (on a plus1 device) to GOS on a Pixel 9. 
Unfortunately I am confronted with the above mentioned error message: "can't mirror to display" "use a different cable and try again".
I am trying to connect a VITURE One Lite. I supposed some setting is incorrect within my GOS (build 2026011301). 
The Viture still works with my PlusOne and it worked, when I tried it on the Pixel 9 before I installed GOS.

What else shall I supply to narrow down the source of this error?

**Comment by sergi0g** [comment]
I'm having the same issue with my Pixel 9a on build 2026011301. I'm trying to connect to a Dell S2722DC monitor with the official cable that came with the phone.

EDIT: I found a different cable and it worked. Ignore this comment.

**Comment by J0ergL** [comment]
@sergi0g : Hidden as resolve means your issue is resolved? Mine is still there :-(
What can one do?
I would like to reopen this issue but that button is inactive.

**Comment by sergi0g** [comment]
> Hidden as resolve means your issue is resolved?

Yep. I found the special cable my display came with and that worked. I had incorrectly assumed that the included charging cable would do the job.

> What can one do?

Are you using the same cable you use for your OnePlus? In any case, if you can get your hands on a cable with verified DP Alt mode support, try that. Maybe also try a different monitor in case a particular combination is problematic.

> I would like to reopen this issue but that button is inactive.

You can always ping a maintainer for that.

**Comment by thestinger** [comment]
> Yep. I found the special cable my display came with and that worked. I had incorrectly assumed that the included charging cable would do the job.

That one doesn't have DisplayPort alternate mode. You need a cable explicitly providing DP alt mode.

**Comment by sergi0g** [comment]
Thanks for confirming!

**Comment by J0ergL** [comment]
> > Hidden as resolve means your issue is resolved?
> 
> Yep. I found the special cable my display came with and that worked. I had incorrectly assumed that the included charging cable would do the job.
> 
> > What can one do?
> 
> Are you using the same cable you use for your OnePlus? In any case, if you can get your hands on a cable with verified DP Alt mode support, try that. Maybe also try a different monitor in case a particular combination is problematic.
Yes, it is the same cable, that was in the box with the glasses and that works with the OnePlus 9 Pro.
I will try to find a monitor with USB C connector.

> 
> > I would like to reopen this issue but that button is inactive.
> 
> You can always ping a maintainer for that.
@maade93791 : Please reopen this issue. Thx
